src/components/ObjectInput.tsx

```diff
@@ -0,0 +1,103 @@
+/* @jsx jsx */
+import { jsx } from '@emotion/core';
+import * as React from 'react';
+
+import { Box, Button, Flex, Text } from '@stoplight/ui-kit';
+
+import { fieldName, IFormtronControl } from '..';
+import { DraftValue } from './DraftValue';
+import { ValidityIndicator } from './ValidityIndicator';
+
+export const ObjectInput: React.FunctionComponent<IFormtronControl> = ({
+  id,
+  value = {},
+  schema,
+  onChange,
+  fieldComponents,
+  selection,
+}) => {
+  // Make shallow copy
+  const items = [...Object.entries(value)];
```

Comment:
Why do we need a shallow copy?

Reply:
```suggestion
  const items = Object.entries(value).slice(0);
```

or also

```suggestion
  const items = Array.from(Object.entries(value));
```

My sensation is that this clearly communicates you're creating a new array — on purpose.

Reply:
I would rather slice.

In theory, both `Array.from` and spread operator used in an array literal initializer don't have to produce the same result as `Array.prototype.slice`

Reply:
I'm fine with slice too.